--- busybox-1.19.3/util-linux/fbset-orig.c	2012-07-01 20:43:04.417996282 +0200
+++ busybox-1.19.3/util-linux/fbset.c	2012-07-01 20:44:55.760288492 +0200
@@ -498,6 +498,18 @@
 		copy_changed_values(&var_old, &var_set);
 		if (options & OPT_ALL)
 			var_old.activate = FB_ACTIVATE_ALL;
+		var_old.red.offset = 10;
+		var_old.red.length = 5;
+		var_old.red.msb_right = 0;
+		var_old.green.offset = 5;
+		var_old.green.length = 5;
+		var_old.green.msb_right = 0;
+		var_old.blue.offset = 0;
+		var_old.blue.length = 5;
+		var_old.blue.msb_right = 0;
+		var_old.transp.offset = 15;
+		var_old.transp.length = 1;
+		var_old.transp.msb_right = 0;
 		xioctl(fh, FBIOPUT_VSCREENINFO, &var_old);
 	}
 
