--- busybox-1.19.3/miscutils/fbsplash-orig.c	2012-07-01 19:24:30.851408689 +0200
+++ busybox-1.19.3/miscutils/fbsplash.c	2012-07-01 19:26:41.584615011 +0200
@@ -109,6 +109,7 @@
  */
 static unsigned fb_pixel_value(unsigned r, unsigned g, unsigned b)
 {
+#if 0
 	if (G.bytes_per_pixel == 2) {
 		r >>= 3;  // 5-bit red
 		g >>= 2;  // 6-bit green
@@ -117,6 +118,8 @@
 	}
 	// RGB 888
 	return b + (g << 8) + (r << 16);
+#endif
+	return ((b & 0xf8) >> 3) | ((g & 0xf8) << 2) | ((r & 0xf8) << 7);
 }
 
 /**
